src/tools/genpolicy/src/registry.rs

```diff
@@ -171,6 +211,110 @@ impl Container {
         }
     }
 
+    pub fn get_gid_from_passwd_uid(&self, uid: u32) -> Result<u32> {
+        if self.passwd.is_empty() {
+            return Err(anyhow!(
+                "No /etc/passwd file is available, unable to parse gids from uid"
+            ));
+        }
+        match parse_passwd_file(self.passwd.clone()) {
```

Comment:
We could look at this in a separate PR, but it seems like the parse_passwd_file parameter should be &String, and then we wouldn't need to clone.

Reply:
Agreed, will at least update the signature of the newly introduced parse_group_file which has the same issue.

Reply:
Honestly, I think it makes sense to simply fix it here.